### PR TITLE
Resolve "Implement DateRangeParam.isDateWithinRange(Date theDate)."

### LINKED ISSUE #1645: Implement DateRangeParam.isDateWithinRange(Date theDate).
The JPA server translates DateRangeParam into SQL. Great! However, I am implementing some hybrid providers, and I'd like to easily match dates against DateRangeParam. I'm going to implement:

`ca.uhn.fhir.rest.param.DateRangeParam`
*  `public boolean isDateWithinRange(Date theDate)`

### PR DESCRIPTION
**Pull request by dmuylwyk**
Closes #1645

### CODE REVIEW

**Comment by dmuylwyk** [comment]
@jamesagnew, please review. Thanks!

File: src/changes/changes.xml [hunk 1]
```diff
@@ -77,6 +77,16 @@
 				An issue with the HAPI FHIR CLI was corrected that prevented the upload of LOINC due to an error
 				regarding the properties file.
 			</action>
+			<action type="add" issue="1645">
```
Comment (jamesagnew) [inline src/changes/changes.xml]: Changelog has moved (I really need to put a note in this file..) can you add your changelog here: 
https://github.com/jamesagnew/hapi-fhir/blob/master/hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/4_2_0/changes.yaml

File: hapi-fhir-base/src/main/java/ca/uhn/fhir/rest/param/DateRangeParam.java [hunk 2]
```diff
@@ -281,7 +281,7 @@ public Date getLowerBoundAsInstant() {
 				case LESSTHAN_OR_EQUALS:
 				case ENDS_BEFORE:
 				case NOT_EQUAL:
-					throw new IllegalStateException("Unvalid lower bound comparator: " + myLowerBound.getPrefix());
+					throw new IllegalStateException("Invalid lower bound comparator: " + myLowerBound.getPrefix());
```
[the remark this answers is not in the document]
Comment (dmuylwyk) [reply hapi-fhir-base/src/main/java/ca/uhn/fhir/rest/param/DateRangeParam.java]: Hehehe.

File: src/changes/changes.xml (hunk 1 above)
Comment (dmuylwyk) [reply src/changes/changes.xml]: Fixed.

**Pull request merged**